cmd/clusterctl/client/tree/discovery_test.go

```diff
@@ -143,6 +143,81 @@ func Test_Discovery(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "Discovery with machine health checks",
+			args: args{
+				discoverOptions: DiscoverOptions{
+					Grouping:                true,
+					ShowMachineHealthChecks: true,
+				},
+				objs: test.NewFakeCluster("ns1", "cluster1").
+					WithControlPlane(
+						test.NewFakeControlPlane("cp").
+							WithMachines(
+								test.NewFakeMachine("cp1"),
+							),
+					).
+					WithMachineDeployments(
+						test.NewFakeMachineDeployment("md1").
+							WithMachineSets(
+								test.NewFakeMachineSet("ms1").
+									WithMachines(
+										test.NewFakeMachine("m1"),
+										test.NewFakeMachine("m2"),
+									),
+							),
+					).
+					WithTopology("top").
+					Objs(),
+			},
+			wantTree: map[string][]string{
+				// Cluster should be parent of InfrastructureCluster, ControlPlane, and WorkerNodes
+				"cluster.x-k8s.io/v1beta1, Kind=Cluster, ns1/cluster1": {
+					"infrastructure.cluster.x-k8s.io/v1beta1, Kind=GenericInfrastructureCluster, ns1/cluster1",
+					"controlplane.cluster.x-k8s.io/v1beta1, Kind=GenericControlPlane, ns1/cp",
+					"virtual.cluster.x-k8s.io/v1beta1, Kind=WorkerGroup, ns1/Workers",
+					"virtual.cluster.x-k8s.io/v1beta1, Kind=MachineHealthCheckGroup, ns1/MachineHealthChecks",
```

Comment:
can you also add some MHC that are children of this virtual node and also verify that in the unit test.

Looks like this unit test does not actually have MHCs in the output tree right now.

Reply:
I've been experimenting with getting the tests and I'm wondering is mhcg node correct here? It's not supposed to be a topology like it's in api:
https://github.com/kubernetes-sigs/cluster-api/blob/ddf405c084d1e6a51815b4934ed55abf5846a954/api/v1beta1/cluster_types.go#L70 In other words it needs to be nested multiple steps to get down to mhcg